include/curl/system.h

```diff
@@ -441,10 +437,22 @@
 #      define CURL_MASK_SCOFFT       CURL_OFF_T_C(0x7FFFFFFFFFFFFFFF)
 #      define CURL_MASK_UCOFFT       CURL_OFF_TU_C(0xFFFFFFFFFFFFFFFF)
 #    else
-#      error "CURL_OFF_T_MAX unexpected value: != 0x7FFFFFFFFFFFFFFF"
+#      if CURL_OFF_T_MAX > CURL_OFF_T_C(0x7FFFFFFFFFFFFFFF)
+#        if CURL_OFF_T_MAX == CURL_OFF_T_C(0x7FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF)
```

Comment:
Isn't this going to break on (old) 32 bit platforms? I mean if the compiler/preprocessor itself doesn't deal with 64 bit values.

Also, remember that this processor magic is run on the host, not necessarily the target. It could at least potentially be a 32 bit preprocessor building code for a 64 bit architecture.

Reply:
> Isn't this going to break on (old) 32 bit platforms? I mean if the compiler/preprocessor itself doesn't deal with 64 bit values.

I don't think so, as I mentioned in the comment that's why I purposely gated the checks in stages rather than just an `#elif`.
~~~c
#else
#      if CURL_OFF_T_MAX > CURL_OFF_T_C(0x7FFFFFFFFFFFFFFF)
#        if CURL_OFF_T_MAX == CURL_OFF_T_C(0x7FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF)
~~~

> Also, remember that this processor magic is run on the host, not necessarily the target. It could at least potentially be a 32 bit preprocessor building code for a 64 bit architecture.

Run on the host using a preprocessor for the target so I don't understand how that would happen. For example in Ubuntu:
~~~
#if LONG_MAX == 0x7FFFFFFF
#error 
#endif
~~~
x86_64-w64-mingw32-gcc -o a a.c